### PR TITLE
SLE-619: Switch to new RulesService

### PR DESCRIPTION
**Pull request by thahnen**
## Summary

Switched to the new RulesService, therefore this PR is more or less a WIP. Updated the background jobs for loading the SonarLintRuleBrowser content to differ based on project / global configuration level.

## Issues

The only issue I'm currently facing is the following: Having a project opened which is bound to my local SQ instance, when analyzing a file of that project and the local SQ instance is down, the `SonarLintBackendService.getEffectiveRuleDetails` will hang in the background job as it awaits its own timeout. It doesn't know for now, that the connection might not be available at all.

### CODE REVIEW

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintBackendService.java [hunk 1]
```diff
@@ -158,6 +163,41 @@ private static String getIdeName() {
     return ideName;
   }
 
+  /**
```
Comment (henryju) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintBackendService.java:166]: I am not convinced of the value of the javadoc here. This is not really an API, and most infos can be guessed from method signature

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/DisplayProjectRuleDescriptionJob.java [hunk 2]
```diff
@@ -19,42 +19,43 @@
...
-  private final ResolvedBinding binding;
+/** Update "web browser" view for the project rule description (maybe context based on connection) in a separate thread */
+public class DisplayProjectRuleDescriptionJob extends AbstractSonarProjectJob {
+  private final ISonarLintProject project;
   private final String ruleKey;
+  private final String contextKey;
   private final SonarLintRuleBrowser browser;
 
-  public AsyncDisplayRuleDescriptionJob(ISonarLintProject project, ResolvedBinding binding, String ruleKey, SonarLintRuleBrowser browser) {
+  public DisplayProjectRuleDescriptionJob(ISonarLintProject project, String ruleKey, String contextKey, SonarLintRuleBrowser browser) {
```
Comment (henryju) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/DisplayProjectRuleDescriptionJob.java:40]: You can mark `contextKey` as Nullable

**Review by henryju — APPROVED** [review]
Minor comments but overall LGTM. See my previous draft PR to fix the IT.

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintBackendService.java (hunk 1 above)
Comment (thahnen) [reply org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintBackendService.java:166]: Yeah I agree, shrunk the comment to only contain a short summary. For me a short summary is necessary, as the method names differ so much, I would play cluedo the next time I encounter that after a long period trying to figure out why the first one contains `Description` and the other one `Details` 👍

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/DisplayProjectRuleDescriptionJob.java (hunk 2 above)
Comment (thahnen) [reply org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/DisplayProjectRuleDescriptionJob.java:40]: Done, also in `SonarLintBackendService.getEffectiveRuleDetails`.

**Comment by thahnen** [comment]
Some ITS are currently failing as I removed the `SonarLintRuleBrowser` HTML content for now, they shall work again when we fix the UI!

**Pull request merged**